src/Output/Output_BasePowerSpectrum.cpp

```diff
@@ -145,16 +145,27 @@ void Output_BasePowerSpectrum( const char *FileName, const long TVar )
       if ( Aux_CheckFileExist(FileName) )
          Aux_Message( stderr, "WARNING : file \"%s\" already exists and will be overwritten !!\n", FileName );
 
+//    data string length
+      const int S_LEN = MAX( abs(atoi(OPT__OUTPUT_DATA_FORMAT+1)), abs(atoi(OPT__OUTPUT_DATA_FORMAT+2)) );
```

Comment:
My current strategy is to read the number before the `.` to determine the length of the string. For example, the format `%20.16e` will give me a length of 20. However, if I only check the string from the `%`, the format `%+-20.16e` (align to the left and also add a `+` for the positive value) will give me a zero string length since `+-` is not a number. Thus, checking `OPT__OUTPUT_DATA_FORMAT+2` is necessary for this case